### PR TITLE
Optimize range search result sort in segcore

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: internal/core/src/common/RangeSearchHelper.cpp [hunk 1]
```diff
@@ -51,73 +57,81 @@ SortRangeSearchResult(DatasetPtr data_set,
 
     // use p_id and p_dist to GenResultDataset after sorted
     auto p_id = new int64_t[topk * nq];
-    memset(p_id, -1, sizeof(int64_t) * topk * nq);
     auto p_dist = new float[topk * nq];
+    std::fill_n(p_id, topk * nq, -1);
```
[the remark this answers is not in the document]
Comment (cydrain) [reply internal/core/src/common/RangeSearchHelper.cpp:61]: here memset() is to set each byte to -1, while fill_n() is to set each int64_t to -1,
using fill_n() can improve the readability.

File: internal/core/src/common/RangeSearchHelper.cpp [hunk 2]
```diff
@@ -51,73 +57,81 @@ SortRangeSearchResult(DatasetPtr data_set,
...
     // The subscript of p_id and p_dist
-    int cnt = 0;
+#pragma omp parallel for
     for (int i = 0; i < nq; i++) {
-        // if RangeSearch answer size of one nq is less than topk, set the capacity to size
-        int size = lims[i + 1] - lims[i];
-        int capacity = topk > size ? size : topk;
+        std::priority_queue<ResultPair, std::vector<ResultPair>, decltype(cmp)>
+            pq(cmp);
+        auto n = std::min<int64_t>(lims[i + 1] - lims[i], topk);
```
[the remark this answers is not in the document]
Comment (cydrain) [reply internal/core/src/common/RangeSearchHelper.cpp]: ok

**Review by congqixia — COMMENTED** [review]
/approve

**Pull request merged**